### DISCUSSION TITLE
Using in conjunction with native SDK?

### DISCUSSION THREAD

**Discussion by iliaskarim** [invertase/react-native-firebase#5360] [Q&A] [answered]
Will the react-native-firebase framework work in conjunction with the native iOS Firebase SDK? For example, to make some calls from native and some from React Native.

**Comment by mikehardy** [comment, marked as answer]
It should yes. Be careful though, react-native-firebase has a strong expectation that *it* is in control of initialization for the apps in firebase-ios-sdk and you will run in to problems if you attempt to configure firebase-ios-sdk using non-react-native-firebase configuration mechanisms.

Instead, if your native code has any execution paths that touch firebase-ios-sdk prior to react-native-firebase starting up and initializing you should call the react-native-firebase initialization machinery to make sure it is set up internally and in a consistent state

This is not really something we support because no one really does it and the ones that do know what they're doing, although not always

related discussion https://github.com/invertase/react-native-firebase/issues/4889

**Reply by iliaskarim** [reply]
@mikehardy thanks for your response. I'd like to start a performance trace on the React Native side and continue it on the native side. However, it doesn't look like that's possible with the framework.

**Reply by mikehardy** [reply]
That's not something I've ever heard of anyone trying, no, but it's software, anything is possible right? Especially given it's open source, there is usually a way to hack *something* in to node_modules code directly to try it, and if we need to make an interface more open or similar that's something that we'll take as a PR historically. I think perf is easier than modules like database or firestore when it comes to brownfield applications because they don't rely nearly as much on settings from plist/json being read and stuffed into the firebase module instances, so init issues may not be as tough to deal with. But you're definitely blazing trail on your own here...
